pkg/scheduler/algorithmprovider/plugins_test.go

```diff
@@ -44,44 +45,52 @@ func TestDefaultConfigExists(t *testing.T) {
 
 func TestAlgorithmProviders(t *testing.T) {
 	for _, pn := range algorithmProviderNames {
-		p, err := factory.GetAlgorithmProvider(pn)
-		if err != nil {
-			t.Errorf("error retrieving '%s' provider: %v", pn, err)
-			break
-		}
-		if len(p.PriorityFunctionKeys) == 0 {
-			t.Errorf("%s algorithm provider shouldn't have 0 priority functions", pn)
-		}
-		for _, pf := range p.PriorityFunctionKeys.List() {
-			if !factory.IsPriorityFunctionRegistered(pf) {
-				t.Errorf("priority function %s is not registered but is used in the %s algorithm provider", pf, pn)
+		t.Run(fmt.Sprintf("providername/%s", pn), func(t *testing.T) {
+			p, err := factory.GetAlgorithmProvider(pn)
+			if err != nil {
+				t.Errorf("error retrieving provider: %v", err)
+				return
 			}
-		}
-		for _, fp := range p.FitPredicateKeys.List() {
-			if !factory.IsFitPredicateRegistered(fp) {
-				t.Errorf("fit predicate %s is not registered but is used in the %s algorithm provider", fp, pn)
+			if len(p.PriorityFunctionKeys) == 0 {
+				t.Errorf("algorithm provider shouldn't have 0 priority functions")
 			}
-		}
+			for _, pf := range p.PriorityFunctionKeys.List() {
+				t.Run(fmt.Sprintf("priorityfunctionkey/%s", pf), func(t *testing.T) {
+					if !factory.IsPriorityFunctionRegistered(pf) {
+						t.Errorf("priority function is not registered but is used in the algorithm provider")
+					}
+				})
+			}
+			for _, fp := range p.FitPredicateKeys.List() {
+				t.Run(fmt.Sprintf("fitpredicatekey/%s", fp), func(t *testing.T) {
+					if !factory.IsFitPredicateRegistered(fp) {
+						t.Errorf("fit predicate is not registered but is used in the algorithm provider")
+					}
+				})
+			}
+		})
 	}
 }
 
 func TestApplyFeatureGates(t *testing.T) {
 	for _, pn := range algorithmProviderNames {
-		p, err := factory.GetAlgorithmProvider(pn)
-		if err != nil {
-			t.Errorf("Error retrieving '%s' provider: %v", pn, err)
-			break
-		}
-
-		if !p.FitPredicateKeys.Has("CheckNodeCondition") {
-			t.Errorf("Failed to find predicate: 'CheckNodeCondition'")
-			break
-		}
-
-		if !p.FitPredicateKeys.Has("PodToleratesNodeTaints") {
-			t.Errorf("Failed to find predicate: 'PodToleratesNodeTaints'")
-			break
-		}
+		t.Run(fmt.Sprintf("providername/%s", pn), func(t *testing.T) {
```

Comment:
This test has an interesting structure. As you've currently written it, you will generate pairs of subtests with the same name. I think it would make sense to replace "providername" with a description of what the two loops are testing. Maybe call them "before_apply" and "after_apply"?

Also, I think the provider name should come first so that the sub-test names are similar to those in TestAlgorithmProviders.